docs/reference/api/platform/CriticalSectionLock.md

```diff
@@ -1,3 +1,18 @@
 ## CriticalSectionLock
 
+CriticalSectionLock class provides a mechanism to access a resource without interruption. With the `lock` API, you can enter critical section with interrupts disabled. The `unlock` API is the exit from critical section, and the state of interrupts is restored upon exit. Nesting of critical section is supported, and interrupts are enabled only when we exit from last nested critical section.
```

Comment:
Do you want to remove mentions the `lock` and `unlock` since these aren't RAII and may be deprecated?

Reply:
I will keep them in 5.6 version as they are relevant. Hopefully we would deprecate by 5.7

Reply:
Sounds good